### ISSUE TITLE
Feature request: Add all available fleets for a given character to the fleet endpoint

### ISSUE THREAD

**Issue by CodeUrself** [ccpgames/esi-issues#887]
# Feature request

Hi!

It came to my attention that the fleet endpoint doesnt list all available fleets for a given character (yet). It would be pretty cool if we could get that endpoint as well, since all other fleet-related endpoints are already implemented.

### Route
`/characters/{character_id}/fleet/` returns the information about the fleet the character is currently in.

`/characters/{character_id}/fleet/all` or similar could return all available fleets for the character.

### Use case
Fleet participation and other fleet management related things can already be implemented with the existing endpoints, however, fleets are always differently named or run under a different person. Identifying the fleet that needs to be tracked/managed by listening all available fleets first would help this in an automated way. 

I think great opportunities for websites that list public NPSI fleets could also emerge. People would be able to see if fleets are being run and who runs it, i'm thinking about Specture Fleet, Zarvox, BjornBee or BombersBar. One quick glance gives everyone the opportunity to find out where they can go to pew pew.
I think it would work great for looser assemblies of more casual groups, newer players or NPSI participants.

Coalition-wide metrics for fleets are currently also heavily favored towards big power blocs, since they have a heavily supporting IT-team that goes above & beyond implementing these metrics manually, by pasting and analyzing D-scans etc. For smaller groups it would be beneficial if that problem could be solved with just an additional ESI endpoint.

### Example returns

```
fleet_id* | integer($int64) returns the fleet ID
fleet_name* | string - Returns the fleet name
fleet_boss | integer($int64) returns the character ID of fleet boss
fleet_location | integer($int64) returns the system ID of the system the fleet is currently in*
```

This would only apply if the fleet location is set to be displayed, otherwise this would return -1.

### Previous XML or CREST route
N/A as far as I know

Thanks in advance!

**Comment by CodeUrself** [comment]
I'd like to add that it seems the endpoint also already exists on CCP's side but hasnt been exposed to ESI (yet, hopefully).

https://i.imgur.com/GLIDost.png
https://i.imgur.com/5g1xG4T.png

**Comment by DaneelTrevize** [comment]
I didn't think there's a way for anyone to advertise a truely public fleet?  
https://i.imgur.com/TCYahJG.png  
It's either to corp, or to those with personal standings. Unless you have some process to automate being make a contact with the FC, how can one list a fleet for all to see and join? [You can't not pick the radio buttons forcing at least Good standing](https://i.imgur.com/A0bDc0a.png).

**Comment by DaneelTrevize** [comment]
What could be beneficial would be separating the read_fleet scope from `/characters/{character_id}/fleet/`. Then those public fleet FCs could potentially grant your proposed NPSI-summarising websites/apps a token to see when they're in a fleet (and FC of it), without having to trust all such apps with the ability to read all details of every fleet they FC (what read_fleet enables atm via /fleets/{fleet_id}/*).

**Comment by CodeUrself** [comment]
> You can easily understand people wanting to advertise that they're running a fleet without giving out the member names, locations and ship composition access.

Thats pretty much what I wrote?

**Comment by DaneelTrevize** [comment]
Then, again, you need to propose a specific new scope that's of this limited listing nature. Or instead request this public fleet listing be under a non-auth'd endpoint, probably under `/fleets/`.  
E.g. `/fleets/public/`, where it maybe takes an optional minimum security status filter, to return those fleets that characters of sufficient sec status can attempt to join.

I still suspect some NPSI FCs might not want their exact fleet/FC location to be public knowledge to all. Perhaps with a 1-5minute cache timer for such a listing endpoint? Still very useful to know what fleets are forming that you might join, but not trivial to know you're at a safe distance from their forward scouts, or completely ready to cyno upon.

**Comment by Aidansavage** [comment]
>Then, again, you need to propose a specific new scope that's of this limited listing nature. Or instead request this public fleet listing be under a non-auth'd endpoint, probably under /fleets/.
E.g. /fleets/public/, where it maybe takes an optional minimum security status filter, to return those fleets that characters of sufficient sec status can attempt to join.


You need neither. Viewing a character's available fleets is _entirely_ under the purview of a `read_fleet` scope for that character. It would never be an unauthed endpoint in the first place, as you need to view those fleets from the POV of a _specific character_ to know whether or not those fleets are available to you.


Now, regarding the _actual_ issue, you could probably skip including the `fleet_id` in the returns as there is no way for you to use that information in the first place without being in the fleet in some fashion.

**Comment by CodeUrself** [comment]
> Now, regarding the actual issue, you could probably skip including the fleet_id in the returns as there is no way for you to use that information in the first place without being in the fleet in some fashion.

Agreed. The fleet ID is unnecessary here unless there would be a method to join such fleet with an ESI POST endpoint.

Thanks for the feedback!

**Comment by NathenSample** [comment]
Daneel, you're missing the entire point of this ticket not to meant hijacking the entire thing to boot. 

The scope of this ticket is that @CodeUrself would like an endpoint that will return to him the contents of the fleet finder in-game.

He doesn't want to advertise his fleets via ESI, make a new issue for that.
He doesn't want to advertise his fleets publicly, talk to game design for that.

**Comment by DaneelTrevize** [comment]
The feature request doesn't address its own use case(s).

Those wishing to track their coalition's fleets (in any detail beyond simply existing) already need their FCs' read_fleet auth'd tokens, this FR doesn't change that or improve on simply iterating over those to assess active fleets.

The case w.r.t. public fleets is fiction, it does not exist in-game via current mechanics.

Yes, there is an in-game fleet search that could have a parallel API endpoint, but no it does not do what is claimed in this FR.

**Comment by DaneelTrevize** [comment]
> A poor or wrong use-case does not invalidate the idea

I can't not reply to this. Reread it back to yourself and seriously question your motives/priorities.

This FR is practically literally useless, it cannot fulfill the stated use cases.  
No NPSI players *can* use the in-game fleet search & listing, thus none do, thus exposing it externally is useless. So it is only of use, even in-game, to those in corps & alliances.  
All corps & alliances can (and I would bet do, but CCP can check) produce the listing from iterating their FCs' read_fleet tokens. Thus this endpoint change would offer nothing new to anyone.  
That's why I've tried to evolve the FR by taking into account at least half a decade of NPSI FCing experience & talking to CSM members, that FCs would need a new very limited read scope before they would trust effectively-public apps to see their current fleets' existance. Then many could benefit, especially across wider community integration such as Twitch chat & Discord bots listing truely public fleets in the future.  
Exposing the in-game fleet finder is picking some rotten low-hanging fruit.

#774 would have far more practical benefit to all those building fleet tools (corp & alliances as well as NPSI) by minimising the token iteration and 404s as described above, and (even ignoring that it has far more votes as per CCP's requested mechanism because NPSI communities were directed to do so to bring their desires to CCP) it's also a longer-standing change request & has duplicates even older.

Yes, I am attacking this idea and now CCP behaviour towards developing it. Not as an anti-personal action, as a lack of apparent reason. Please feel free to do likewise and enlighten me.

**Comment by a-tal** [comment]
Just because it's not your use case doesn't mean it's an invalid use case, nor does it make it "rotten low-hanging fruit".

You even say the use case yourself, `So it is only of use, even in-game, to those in corps & alliances.` That's a massive chunk of the player base. I can personally see value in this, as it could allow for easier/no pings out of game, and thus more player interaction.

Your comments here don't seem to add any value to the issue at hand and could probably be summarized with a single 'thumbs down' on the parent comment. I get that you're passionate about fully public fleets, but you really need to contain that passion to issues relating to it.

**Comment by DaneelTrevize** [comment]
"it could allow for easier/no pings out of game" Are we talking about an out-of-game API endpoint or what?!

No corp/alliance players that want to find available fleets out-of-game can't have that fulfilled by the existing endpoints (remember they need someone to make an app for it, they aren't getting this via some polished CCP website, unless you're claiming they're use ESI/ui/ to do so...), and those endpoints are only hampered by that which #774 addresses. This FR does not practically help that, and helps no one else.

**Comment by NyxVxyN** [comment]
All I can see is this being automated spying, yet adding to multiple of the nullsec entities bots capabilities...

**Comment by Aidansavage** [comment]
>All I can see is this being automated spying, yet adding to multiple of the nullsec entities bots capabilities...

And how would you explain that? Even beyond the fact that if CCP has reason to believe there's any malicious activity happening with an app, they can disable it? You also say it like people dont already have spies in other entities that can get this information to begin with.

**Comment by a-tal** [comment]
approved by game design

**Comment by DaneelTrevize** [comment]
In commemoration, I drew you all a picture in Paint3D (turns out it easily has terrible font rendering).

https://i.imgur.com/l4xYeO3.png

**Comment by a-tal** [comment]
the discussion here was all sorts of off-topic. let's try to stay focused on the issue at hand in the future.

apologies, you cannot leave a 👍 reaction on this issue anymore

**Comment by jowrjowr** [comment]
open for business don't shit it up too hard

**Comment by Flupster** [comment]
Did this issue ever come to fruition? if not could we get a second look and potentially have this endpoint available, as it stands afaik you still have to guess the fleet boss which can be problematic and bounced failed requests until you figure out the boss
